pkg/chunk/grpc/storage_client.go

```
@@ -0,0 +1,124 @@
+package grpc
+
+import (
+	"context"
+	"io"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+)
+
+type StorageClient struct {
+	cfg        Config
+	schemaCfg  chunk.SchemaConfig
+	client     GrpcStoreClient
+	connection *grpc.ClientConn
+}
+
+// NewStorageClient returns a new StorageClient.
+func NewStorageClient(cfg Config, schemaCfg chunk.SchemaConfig) (*StorageClient, error) {
+	grpcClient, conn, err := connectToGrpcServer(cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+	client := &StorageClient{
+		cfg:        cfg,
+		schemaCfg:  schemaCfg,
+		client:     grpcClient,
+		connection: conn,
+	}
+	return client, nil
+}
+
+func (s *StorageClient) Stop() {
+	_, err := s.client.Stop(context.Background(), &Nothing{})
+	if err != nil {
+		return
```

Never suppress errors. Please log it.